saleor/graphql/checkout/mutations.py

```diff
@@ -1278,8 +1293,30 @@ def perform_mutation(
                 raise e
 
             manager = info.context.plugins
-            lines = fetch_checkout_lines(checkout)
-            validate_variants_in_checkout_lines(lines)
+            lines, unavailable_variant_pks = fetch_checkout_lines(checkout)
```

Comment:
I see at least 4 places where we're using fetch_checkout_lines to check if variants are unavailable. So my question is - wouldn't be good to create two separate functions:
- fetch_checkout_info - function that fetches all data as it was done previously OR function with new input + changes in many files
- fetch+validation - this can be called in different places and will give us, a single point for handling a validation of checkout's lines.  Then we could call this function in all places where valid variants are important to correctly process a checkout

Reply:
@maarcingebala @fowczarek

Reply:
IMO we could merge this PR and add a task with your proposition idea.

Reply:
Agree with @fowczarek here.

Reply:
@fowczarek @maarcingebala Okay.  But we need to keep in mind that we have now 4 places where this validation is made. Any change or extending it will require adding/changing it in 4 places.